source/gfoidl.Base64/Internal/Base64Encoder/Base64Encoder.Decode.netcoreapp.cs

```diff
@@ -39,21 +40,26 @@ public partial class Base64Encoder
 
             ref byte destBytes = ref MemoryMarshal.GetReference(data);
 
+            if (maxSrcLength < 24)
+                goto Scalar;
+
             if (Avx2.IsSupported && maxSrcLength >= 45)
             {
                 Avx2Decode(ref src, ref destBytes, maxSrcLength, destLength, ref sourceIndex, ref destIndex);
 
                 if (sourceIndex == srcLength)
                     goto DoneExit;
             }
-            else if (Ssse3.IsSupported && maxSrcLength >= 24)
+
+            if (Ssse3.IsSupported && (maxSrcLength >= (int)sourceIndex + 24))
             {
                 Ssse3Decode(ref src, ref destBytes, maxSrcLength, destLength, ref sourceIndex, ref destIndex);
 
                 if (sourceIndex == srcLength)
                     goto DoneExit;
             }
 
+        Scalar:
```

Comment:
Or use just a plain `if` instead of the `goto`...
Same on other places (encoder as well).